### ISSUE TITLE
Make it easy to find the narrative docs from a link to class documentation

### ISSUE THREAD

**Issue by mhvk** [astropy/astropy#15516]
### What is the problem this feature will solve?

As noted by @taldcroft in https://github.com/astropy/astropy/pull/15231#discussion_r1365285831, in the context of a what's new entry for using `Masked` inside `Time`:
> This highlights a systemic problem (IMO) in our docs, which is typically linking to a class docstring instead of the narrative docs that explain how to use that class. Basically [Masked](https://docs.astropy.org/en/stable/api/astropy.utils.masked.Masked.html#astropy.utils.masked.Masked) is useless and frustrating.

I think this is indeed a problem. In principle, one could add links to the narrative docs in the docstring, though perhaps it is easy to make a general solution? Indeed, if one follows the link, to https://astropy.readthedocs.org/en/latest/api/astropy.utils.masked.Masked.html#astropy.utils.masked.Masked, one indeed does not immediately see something. Just scroll up a tiny bit, though, and one gets
![image](https://github.com/astropy/astropy/assets/2789820/85783e74-3860-4f51-a509-f82a7d6c087e)

So, it does give the section the API is defined in.

For `Time`, it is a bit less good:
![image](https://github.com/astropy/astropy/assets/2789820/eed7a824-a924-4f1b-b4c6-e472245d62d9)

This is because the file defines an explicit API section and that is all one can click on.

### Describe the desired outcome

Links to the main astropy classes should go to documentation that has an obvious link to the narrative docs.

In what's-new, we probably should more generally link to the narrative docs instead of to classes.

**Comment by pllim** [comment]
Probably more chance of getting this resolved if you narrow the scope down a little. Anything that requires refactoring doc for the whole package is unlikely to get done in a short timescale.

**Comment by taldcroft** [comment]
Good point. At some level the simplest thing is a policy to include links to narrative docs in "major" class definitions. It turns out this is already being done for `Time`, but I might move the reference forward to make it even more visible.
```
    """
    Represent and manipulate times and dates for astronomy.

    See http://docs.astropy.org/en/stable/time/ for more information.

    A `Time` object is initialized with one or more times in the ``val``
    argument.  The input times in ``val`` must conform to the specified
    ``format`` and must correspond to the specified time ``scale``.  The
    optional ``val2`` time input should be supplied only for numeric input
    formats (e.g. JD) where very high precision (better than 64-bit precision)
    is required.
...
```

**Comment by mhvk** [comment]
@pllim - the plan definitely was not for a major overhaul, just to trigger thinking how in a *simple* way we can improve the situtation. My sense is that this can auto-generated, by having the docs show a link to the page in which the API is defined (e.g., under the `Bases` link.

Though an explicit link in the docstring has the advantage that one also sees it in ipython, etc. In that case, we have to make sure there is a full link (as in @taldcroft's example) rather than just `astropy.time`, otherwise it is not copy-pastable. And I think this is necessary only for the major classes.

**Comment by pllim** [comment]
I am not sure about auto-generated. I think that would be a pain to be made generic enough for all the sub-packages. Much easier solution is for devs to go link to subpackage user doc inside API docstring, preferably using `:ref:` tag.

**Comment by mhvk** [comment]
> preferably using `:ref:` tag.

If we go for links in docstrings, then I think for people doing things like `Time?` in ipython, we really have to include a full `https` link rather than something that works only inside the sphinx-generated html output.

**Comment by pllim** [comment]
If you hardcode the URL, you will get unsync-ed doc in PR builds, RCs, and so on. 🤷

**Comment by taldcroft** [comment]
> If you hardcode the URL, you will get unsync-ed doc in PR builds, RCs, and so on. 🤷

I was waiting for this. :smile: Perfect is the enemy of good, and here we just want to address the 99% use case of a major class and allow users to quickly get back to the stable narrative docs.

I also agree with @mhvk's comments that the ipython text docs are important so hardcoded https links (even though they are theoretically subject to breakage) would be preferred. In practice something like https://docs.astropy.org/en/stable/time/ is not going to change any time soon.
